lightning/src/chain/keysinterface.rs

```diff
@@ -184,8 +191,9 @@ impl ChannelKeys for InMemoryChannelKeys {
 	fn htlc_base_key(&self) -> &SecretKey { &self.htlc_base_key }
 	fn commitment_seed(&self) -> &[u8; 32] { &self.commitment_seed }
 
-	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, channel_funding_redeemscript: &Script, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>) -> Result<(Signature, Vec<Signature>), ()> {
+	fn sign_remote_commitment<T: secp256k1::Signing>(&self, channel_value_satoshis: u64, channel_funding_redeemscript: &Script, feerate_per_kw: u64, commitment_tx: &Transaction, keys: &TxCreationKeys, htlcs: &[&HTLCOutputInCommitment], to_self_delay: u16, secp_ctx: &Secp256k1<T>, redeem_scripts: &Vec<Script>, remote_per_commitment_point: &PublicKey) -> Result<(Signature, Vec<Signature>), ()> {
 		if commitment_tx.input.len() != 1 { return Err(()); }
+		if commitment_tx.output.len() != redeem_scripts.len() { return Err(()); }
```

Comment:
We should walk the full list and check that they hash to the output script_pubkey, no?

Reply:
Yes, that would be best.  I wasn't sure if we want to have more or less validation right off the bat.  Happier with more, so will add this check now.

Reply:
Oops, sorry, would be better to put this in EnforcingChannelKeys since otherwise we're just needlessly slowing down InMemoryChannelKeys when we "know" that it is fine (mostly cause EnforcingChannelKeys tests it in fuzz targets :p).

Reply:
done